src/libpriv/rpmostree-core.c

```diff
@@ -1930,7 +1930,57 @@ rpmostree_context_prepare (RpmOstreeContext *self,
   g_autoptr(GHashTable) local_pkgs_to_install =
     g_hash_table_new_full (g_str_hash, g_str_equal, NULL, g_object_unref);
 
-  /* Handle packages to replace; only add them to the sack for now */
+  /* We download URI packages and import them into the pkgcache first so we can
+  add it to the local packages to install */
+  g_autoptr(GUnixFDList) fd_list = g_unix_fd_list_new ();
+  for (char **it = pkgnames; it && *it; it++)
```

Comment:
Hmm, were you able to try out the `cached-packages` approach we discussed? Should mostly be about taking this diff hunk here and putting it on the compose side, and then feeding the NEVRAs that the importer returns into the `cached-packages` field.

Reply:
> Hmm, were you able to try out the `cached-packages` approach we discussed? Should mostly be about taking this diff hunk here and putting it on the compose side, 

@jlebon Can you clarify on what you mean by putting it on the compose side?


> and then feeding the NEVRAs that the importer returns into the `cached-packages` field.

Should I create another `cached-packages` array and add the NEVRAS in the loop at https://github.com/projectatomic/rpm-ostree/pull/1847/files#diff-07963b1a5922690400970963652cbe3bR1967 ?





> Hmm, were you able to try out the `cached-packages` approach we discussed? Should mostly be about taking this diff hunk here and putting it on the compose side, and then feeding the NEVRAs that the importer returns into the `cached-packages` field.